firebase_admin/ml.py

```diff
@@ -633,6 +647,183 @@ def as_dict(self, for_upload=False):
         return {'automlModel': self._auto_ml_model}
 
 
+class TFLiteManagedUploadSource(TFLiteModelSource):
+    """TFLite model source representing a TFLite model uploaded in the console. (Output only)."""
+    def __init__(self):
+        pass
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    @property
+    def managed_upload(self):
+        return True
+
+    def as_dict(self, for_upload=False):
+        # This class is output only, you can't create
+        # a model from a Managed Upload Source since that
+        # indicates it was created through the console, and
+        # this is not the console.
+        # (the backend will ignore the managed_upload field)
+        return {'managedUpload': True}
+
+
+class CoreMlFormat(ModelFormat):
+    """Model format representing a Core ML model.
+
+    Args:
+      model_source: A CoreMlModelSource sub class. Specifies the details of the model source.
+    """
+    def __init__(self, model_source=None):
+        self._data = {}
+        self._model_source = None
+
+        if model_source is not None:
+            self.model_source = model_source
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create an instance of the object from a dict."""
+        data_copy = dict(data)
```

Comment:
I'm not sure how to do the coreml_format._data = data_copy then. Init just takes care of the source. The format doesn't exist until after CoreMlFormat(...) ... But the init is required to pop certain things off the data_copy. I think this needs to stay here? Am I missing something?

Reply:
It's a little confusing. It seems we want `coreml_format._data` to be a dictionary that doesn't contain the source information. I'd think it shouldn't matter (given `_data` is not directly exposed). Why is this a requirement? Is it because of `__eq__`?